apps/api/src/app/integrations/dtos/create-integration-request.dto.ts

```diff
@@ -1,11 +1,27 @@
-import { IsBoolean, IsDefined, IsEnum, IsString, ValidateNested } from 'class-validator';
+import { IsBoolean, IsDefined, IsEnum, IsMongoId, IsOptional, IsString, ValidateNested } from 'class-validator';
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+import { Type } from 'class-transformer';
 import { ChannelTypeEnum, ICreateIntegrationBodyDto } from '@novu/shared';
+
 import { CredentialsDto } from './credentials.dto';
-import { ApiProperty } from '@nestjs/swagger';
-import { Type } from 'class-transformer';
 
 export class CreateIntegrationRequestDto implements ICreateIntegrationBodyDto {
-  @ApiProperty()
+  @ApiPropertyOptional({ type: String })
+  @IsOptional()
+  @IsString()
+  name?: string;
+
+  @ApiPropertyOptional({ type: String })
+  @IsOptional()
+  @IsString()
+  identifier?: string;
+
+  @ApiPropertyOptional({ type: String })
+  @IsOptional()
+  @IsMongoId()
+  _environmentId?: string;
```

Comment:
The new optional fields: `name, identifier, _environmentId`. The `name` and `identifier` will be auto-generated if not provided. The `_environmentId` if not provided will be picked from the user token.